torch/testing/_internal/distributed/rpc/jit/rpc_test.py

```diff
@@ -1038,6 +1038,32 @@ def callback(fut):
 
         self.assertEqual(fut.wait(), torch.ones(n, n) + 1 + num_cbs)
 
+    @dist_init
+    def test_add_done_callback(self):
+        callback_called = None
+
+        def callback(fut):
+            nonlocal callback_called
+            callback_called = fut.wait() * 2
+
+        future = rpc.rpc_async(
+            worker_name((self.rank + 1) % self.world_size),
+            script_fork_wait_udf,
+            args=(torch.ones(2),),
+        )
+
+        future.add_done_callback(callback)
+        future_then = future.then(lambda _: True)
+
+
+        self.assertEqual(callback_called, None)
```

Comment:
The callbacks should run when the `markCompleted` method is called on future, usually by the thread in the thread pool that processes the RPC response. Given this, isn't it theoretically possible the callback would've ran here and this assert could be flaky?

Reply:
yep you're right, thanks for pointing that out. I think when I was writing corresponding tests between `test_futures.py` and `rpc_test.py` I missed the fact that "create the future" and "mark the future as complete" aren't split into two deterministic, separate steps. I'm just going to delete this assert, since I don't think that it adds much to the test (I could also move it to before the call to `add_done_callback` but I don't think that really adds anything).